packages/containers/src/DeleteResource/sagas.js

```diff
@@ -75,17 +74,26 @@ export function* deleteResourceValidate(
 			yield put({
 				type: deleteResourceConst.DIALOG_BOX_DELETE_RESOURCE_SUCCESS,
 				model: {
+					id: safeId,
 					labelResource: resource.get('label') || resource.get('name', ''),
 				},
 			});
+			yield put(
+				cmf.actions.collections.mutate(resourceLocator, {
+					delete: [safeId],
+				}),
+			);
 		}
-		yield call(redirect, action);
+		const url = get(action, 'data.model.redirectUrl');
+		yield call(redirect, url);
 	}
 }
 
 export function* deleteResourceCancel() {
 	const action = yield take(deleteResourceConst.DIALOG_BOX_DELETE_RESOURCE_CANCEL);
-	yield call(redirect, action);
+	const url =
+		get(action, 'data.model.onCancelRedirectUrl') || get(action, 'data.model.redirectUrl');
```

Comment:
Could be simplified by using `get` default value.
`get(action, 'data.model.onCancelRedirectUrl', get(action, 'data.model.redirectUrl'))`